### ISSUE TITLE
Canceled order + deleted shipping address assigned to a new, wrong customer

### ISSUE THREAD

**Issue by aziule** [PrestaShop/PrestaShop#11086]
**Describe the bug**
After canceling an order and deleting the client information, a new order was passed by a new customer, but the previously canceled order was assigned to the new customer, and the new customer sees a new shipping address being assigned to him and corresponding to the previously deleted customer's shipping address.

The steps described below are what happened on my live website.

**To Reproduce**
Steps to reproduce the behaviour:
1. Order a product without creating an account
2. In the back-end, set the order as canceled and delete the client (allow him to create an account later)
3. A new client creates an account and makes an order
4. Problems:
* The previously cancelled order will be assigned to the new customer
* The new customer will have an extra, unwanted shipping address assigned to him, equal to the deleted customer's shipping address

**Additionnal information**
PrestaShop version: 1.7.4.3
PHP version: 7.0.32

I do not currently have access to my dev env so I can't reproduce

**Comment by khouloudbelguith** [comment]
Hi @Aziule,

Thanks for your report.
I did not manage to reproduce the issue with the last version 1.7.4.3.
I attached a video record.
https://drive.google.com/file/d/1GUww7Ooeb9yHM9Aw0aiMQUHAXhooWLq0/view
Thanks to check & feedback.

**Comment by khouloudbelguith** [comment]
Hi @Aziule,

Since we had no news from you for more than 30 days, I'll close this ticket. Feel free to open another one if you can give specific details.
Thanks!

**Comment by dpalauzov** [comment]
Hi Guys, 

I've experienced the same issue.

1. I did a test order, without registering the account
2. Deleted the account and cancelled the orders
3. New user created account and 3 of those orders were assigned to him with my test shipping addresses.

Have a look at the screenshots and the dates of the ordes and the date of the registered user.
![orders date](https://user-images.githubusercontent.com/43937843/58991069-13775800-87f0-11e9-8c37-3db1c3c82bbe.png)
![register](https://user-images.githubusercontent.com/43937843/58991071-13775800-87f0-11e9-97fc-c4907f6340b7.png)

Thanks,
Dimitar
